src/database.rs

```diff
@@ -132,4 +134,99 @@ impl dyn Database {
         let views = self.zalsa().views();
         views.downcaster_for().downcast(self)
     }
+
+    /// Returns information about any Salsa structs.
+    pub fn structs_info(&self) -> Vec<IngredientInfo> {
```

Comment:
maybe call this `ingredients`?

Reply:
It's not all ingredients, it's only tracked/interned/input structs.

Reply:
Queries are also ingredients but they're split out from this method. 

Internally there's also a bunch of other things that we call ingredients, but I'm not sure that's relevant for this API (tracked struct fields, etc.).